### PR TITLE
Dashboard backend

### PR DESCRIPTION
**Pull request by BaileyMcCauslin**
Resource submission, resource review, and search analytics.

### CODE REVIEW

**Review by nicolesylvester — APPROVED** [review]

File: lib/createResource.dart [hunk 1]
```diff
@@ -208,6 +215,11 @@ class _CreateResourceState extends State<CreateResource> {
         'createdTime': FieldValue.serverTimestamp(),
       };
 
+      // Submit to admin submission if object not null
+      if(userSubmission != null) {
+        userSubmission?.submittedResource(resourceName, resourceType);
+      }
```
Comment (JavadocMD) [inline lib/createResource.dart:221]: I believe you don't need the `if` here -- `userSubmission?.submittedResource(...)` should skip calling the method if the userSubmission object is null.

Comment (JavadocMD) [reply lib/createResource.dart:221]: `?.` is part of a family of null-aware operations supported by Dart.

File: lib/view_resource/filter.dart [hunk 2]
```diff
@@ -122,6 +125,53 @@ class ResourceFilter {
   }
 }
 
+// Resonsible for uploading searches and filter options
+class HomeAnalytics {
```
Comment (JavadocMD) [inline lib/view_resource/filter.dart:129]: Should we group all of the analytics classes into one module? I think HomeAnalytics and UserAnalytics could be co-located.

File: lib/view_resource/filter.dart [hunk 3]
```diff
@@ -5,6 +5,9 @@ import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:flutter/material.dart';
 import 'package:web_app/events/schedule.dart';
 
+// Time
+import '../common.dart';
```
Comment (JavadocMD) [inline lib/view_resource/filter.dart:9]: I was thinking about standardizing towards absolute imports and away from relative imports -- i.e., `import 'package:web_app/common.dart';` One advantage being you don't have to change a module's imports if you ever move a module into or out of a folder. What do you think?

**Review by JavadocMD — COMMENTED** [review]
See inline comments. These are fairly minor -- mostly code organization and standards -- the general approach seems good.

**Review by JavadocMD — APPROVED** [review]

**Comment by JavadocMD** [comment]
Good to squash-merge to main.

**Pull request merged**